common/src/main/java/com/microsoft/identity/common/internal/commands/InteractiveTokenCommand.java

```diff
@@ -39,21 +43,53 @@
 @EqualsAndHashCode(callSuper = true)
 public class InteractiveTokenCommand extends TokenCommand {
     private static final String TAG = InteractiveTokenCommand.class.getSimpleName();
+    //Normally all tasks have an affinity unless configured explicitly for multi-window support to not have one
+    private boolean mHasTaskAffinity = true;
+    private int mTaskId = 0;
 
     public InteractiveTokenCommand(@NonNull final InteractiveTokenCommandParameters parameters,
                                    @NonNull final BaseController controller,
                                    @SuppressWarnings(WarningType.rawtype_warning) @NonNull final CommandCallback callback,
                                    @NonNull final String publicApiId) {
         super(parameters, controller, callback, publicApiId);
+        checkAndRecordTaskInformation(parameters);
     }
 
     public InteractiveTokenCommand(@NonNull InteractiveTokenCommandParameters parameters,
                                    @NonNull List<BaseController> controllers,
                                    @SuppressWarnings(WarningType.rawtype_warning) @NonNull CommandCallback callback,
                                    @NonNull final String publicApiId) {
         super(parameters, controllers, callback, publicApiId);
+        checkAndRecordTaskInformation(parameters);
     }
 
+    private void checkAndRecordTaskInformation(@NonNull final InteractiveTokenCommandParameters parameters){
+        final String methodName = ":checkAndRecordTaskInformation";
+        final Context applicationContext = parameters.getAndroidApplicationContext();
+        final PackageManager packageManager = applicationContext.getPackageManager();
+        try {
+            final ActivityInfo startActivityInfo = packageManager.getActivityInfo(parameters.getActivity().getComponentName(), 0);
+            if(startActivityInfo.taskAffinity == null){
+                mHasTaskAffinity = false;
+                mTaskId = parameters.getActivity().getTaskId();
+            }
+        } catch (final PackageManager.NameNotFoundException e) {
+            Logger.warn(
+                    TAG + methodName,
+                    "Unable to get ActivityInfo for activity provided to start authorization."
+            );
+        }
+    }
+
+    public boolean getHasTaskAffinity(){
```

Comment:
```suggestion
    public boolean hasTaskAffinity(){
```

Reply:
To be honest I don't care what the standard is... as long as we're all following it.  Unfortunately we have a diversity of things now in the code.